swift/Tests/SwiftNoosphereTests/NoosphereTests.swift

```diff
@@ -407,4 +407,189 @@ final class NoosphereTests: XCTestCase {
         ns_sphere_fs_free(sphere_fs)
         ns_free(noosphere)
     }
+    
+    func testSettingAndGettingAPetname() throws {
+        let noosphere = ns_initialize("/tmp/foo", "/tmp/bar", nil, nil)
+
+        ns_key_create(noosphere, "bob", nil)
+
+        let sphere_receipt = ns_sphere_create(noosphere, "bob", nil)
+
+        let sphere_identity_ptr = ns_sphere_receipt_identity(sphere_receipt, nil)
+        let sphere = ns_sphere_fs_open(noosphere, sphere_identity_ptr, nil)
+        
+        ns_string_free(sphere_identity_ptr)
+        ns_sphere_receipt_free(sphere_receipt)
+        
+        ns_sphere_petname_set(noosphere, sphere, "alice", "did:key:alice", nil)
+        ns_sphere_fs_save(noosphere, sphere, nil, nil)
+        
+        let identity_ptr = ns_sphere_petname_get(noosphere, sphere, "alice", nil)
+        let identity = String.init(cString: identity_ptr!)
+        
+        assert(identity == "did:key:alice")
+        
+        ns_string_free(identity_ptr)
+        ns_sphere_fs_free(sphere)
+        ns_free(noosphere)
+    }
+
+    func testGetAllPetnamesInASphere() throws {
```

Comment:
This test (and the next one) are _extremely_ helpful for understanding the consumption side of the story 🙏